checks.d/openstack.py

```diff
@@ -137,10 +139,32 @@ def from_config(cls, init_config, instance_config):
         auth_scope = cls.get_auth_scope(instance_config)
         identity = cls.get_user_identity(instance_config)
 
+        exception_msg = None
         try:
-            auth_resp = cls.request_auth_token(auth_scope, identity, keystone_server_url, ssl_verify)
+            auth_resp = cls.request_auth_token(auth_scope, identity, keystone_server_url, ssl_verify, proxy_config)
         except (requests.exceptions.HTTPError, requests.exceptions.Timeout, requests.exceptions.ConnectionError):
-            raise KeystoneUnreachable()
+            exception_msg = "Failed kaystone auth with identity:{id} scope:{scope} @{url}".format(
+                id=identity,
+                scope=auth_scope,
+                url=keystone_server_url)
+
+        if exception_msg:
+            try:
+                identity['password']['user']['domain']['name'] = identity['password']['user']['domain'].pop('id')
```

Comment:
Sorry my bad, completely missed the test cases, that's exactly what I was talking about, as long as we have them, accessing those dict is perfectly fine as it is now.